ztp: CNF-7426: Adding tests for pgt2acmpg translator

edcdavid opened the pull request:
This PR is doing the following:
- introducing a PGT to ACM Gen translator tool, pgt2acm (see https://github.com/test-network-function/pgt2acm). The source code for the translator is saved under zip/pgt2am.
- a version.json is added to indicate the version of the pgt2acm tool to pull. Updates to the code should be merged when updating the released version
- adding a CI test for the PGT to ACM Gen translator (https://github.com/test-network-function/pgt2acm). The test translates the reference PGT templates to ACM Gen.  Then the generated ACM Gen templates are used to generate ACM policies. In the last step, the Generated ACM policies from PGT and PGT converted to ACM Gen  are  compared with the Diff tool (policy-object-template-diff). If ACM Policies generated from converted PGT templates match ACM policies from PGT templates, the test is a success.  
- improved documentation to help use ACM policies

sabbir-47 commented:
/ok-to-test

edcdavid commented:
/retest

edcdavid commented:
/retest

edcdavid commented:
/retest

edcdavid commented:
/retest

edcdavid commented:
/cc @sabbir-47  @irinamihai  @lack

sabbir-47 commented on ztp/gitops-subscriptions/Makefile:
```diff
@@ -45,4 +49,41 @@ gen-files: build $(KUSTOMIZE)
+pgt2acm-clean:
+	rm -rf ../tools/pgt2acm
+
+pgt2acm-get: pgt2acm-clean
```
If the tool is moving under ztp/tools, is there any need to clone it?

edcdavid replied:
Fixed. I was planning to copy the pgt2acm project from the original repo and periodically synchronize with a script to avoid issues with having multiple go projects under the same Github repo.  I removed all that synchronization logic.

edcdavid commented:
/retest-required

edcdavid commented:
/cc fontivan

edcdavid commented:
/jira refresh

edcdavid commented:
/bugzilla refresh

edcdavid commented:
/bugzilla refresh

fontivan commented on ztp/resource-generator/Containerfile:
```diff
@@ -36,6 +42,11 @@ RUN chown -R 1001:1001 $ZTP_HOME
+COPY --from=builder $BUILDER_ZTP/tools/pgt2acm/pgt2acm /usr/bin
+COPY --from=builder /usr/bin/tree /usr/bin
+COPY --from=builder  $BUILDER_ZTP/tools/pgt2acm/kustomize /kustomize-pgt2acm
+RUN tree /kustomize-pgt2acm
```
I think this will require a midstream change, will discuss with @rauhersu

edcdavid commented:
/bugzilla refresh

fontivan commented on ztp/resource-generator/Containerfile:
```diff
@@ -36,6 +42,11 @@ RUN chown -R 1001:1001 $ZTP_HOME
+# Copy pgt2acm script
+COPY --from=builder $BUILDER_ZTP/tools/pgt2acm/pgt2acm /usr/bin
+COPY --from=builder /usr/bin/tree /usr/bin
+COPY --from=builder  $BUILDER_ZTP/tools/pgt2acm/kustomize /kustomize-pgt2acm
```
nit: extra space between `--from=builder` and `$BUILDER_ZTP` on line 48 only

edcdavid replied:
fixed

edcdavid commented:
/retest

edcdavid commented:
/bugzilla refresh

irinamihai commented on ztp/gitops-subscriptions/argocd/example/acmpolicygenerator/README.md:
```diff
@@ -1,5 +1,78 @@
-## ACM PolicyGenerator
-The [ACM PolicyGenerator](https://github.com/stolostron/policy-generator-plugin) examples under [here](https://github.com/openshift-kni/cnf-features-deploy/tree/master/ztp/gitops-subscriptions/argocd/example/acmpolicygenerator) are defining the DU profile policies using [ACM PolicyGenerator reference API](https://github.com/stolostron/policy-generator-plugin/blob/main/docs/policygenerator-referen ...
+# ACM PolicyGenerator 
+The ACM Policy Generator Custom Resource (CRs) is the official template mechanism used to generate ACM policies. It is replacing Policy Generator Templates or PGT.
```
Maybe something along the lines of 
``It is the recommended alternative to Policy Generator Templates (PGT).`` ?

edcdavid replied:
fixed

rauhersu commented on ztp/resource-generator/Containerfile:
```diff
@@ -15,6 +16,11 @@ WORKDIR $PGT_ROOT
+WORKDIR $PGT2ACM_ROOT
+RUN go mod vendor
+RUN make build
+RUN yum install tree
```
do we need 'yum -y' in case it might ask about y/n ? 

Cc: @fontivan

fontivan replied:
That is probably a good idea

rauhersu replied:
Btw, if I am not wrong, this 'tree' command has nothing to do with a builder stage. Wouldn't it be more appropriate to install this as part of the runtime stage (FROM ubi) ?

edcdavid replied:
fixed, added yum -y

rauhersu commented:
/hold

We need to perform midstream changes to adapt these upstream changes. Midstream MR is underway, as soon as this review finishes off, we will include the latest changes (if any) to proceed.

irinamihai commented on ztp/gitops-subscriptions/argocd/example/acmpolicygenerator/README.md:
```diff
@@ -1,5 +1,78 @@
+
+| PGT patch | ACM Generator patch |
+|-----------|---------------------|
+| Uses Strategic merge [link](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-api-machinery/strategic-merge-patch.md)|PGT patches work by replacing variables with their values as defined by the patch|
```
I'm wondering if it would be better to link to the kustomize documentation instead.
https://github.com/kubernetes-sigs/kustomize/blob/master/examples/patchMultipleObjects.md

edcdavid replied:
fixed

irinamihai commented on ztp/gitops-subscriptions/argocd/example/acmpolicygenerator/README.md:
```diff
@@ -1,5 +1,78 @@
+|-----------|---------------------|
+| Uses Strategic merge [link](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-api-machinery/strategic-merge-patch.md)|PGT patches work by replacing variables with their values as defined by the patch|
+|Relies only on patching, no embedded variable substitution|Overwrites values defined in patch|
+|Does not support merging lists, only replace (missing support for schema)|Substitute variables defined in source CR with values defined in patch (for example $name)|
```
```suggestion
|Does not support merging lists, only replace (missing support for OpenAPI schema)|Substitute variables defined in source CR with values defined in patch (for example $name)|
```

edcdavid replied:
fixed

irinamihai commented on ztp/gitops-subscriptions/argocd/example/acmpolicygenerator/README.md:
```diff
@@ -1,5 +1,78 @@
+| Uses Strategic merge [link](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-api-machinery/strategic-merge-patch.md)|PGT patches work by replacing variables with their values as defined by the patch|
+|Relies only on patching, no embedded variable substitution|Overwrites values defined in patch|
+|Does not support merging lists, only replace (missing support for schema)|Substitute variables defined in source CR with values defined in patch (for example $name)|
+|Requires additional directives ($patch: replace) in patch to merge content that does not follow a schema (ptp plugins object)| Can overwrite the name and namespace defined in source CR reference|
```
```suggestion
|Requires additional directives ($patch: replace) in patch to merge content that does not follow a schema (PTP plugins object)| Can overwrite the name and namespace defined in source CR reference|
```

irinamihai commented on ztp/resource-generator/entrypoints/entrypoints:
```diff
@@ -49,3 +49,17 @@ generator
+/pgt2acm
+-------------------
+
+pgt2acm translate a base policygen template directory structure to
```
typo: `translates`

edcdavid replied:
fixed

irinamihai commented on ztp/tools/pgt2acm/main.go:
```diff
@@ -0,0 +1,357 @@
+
+	preRenderPatchKindList, preRenderSourceCRList := processFlags(inputFile, outputDir, preRenderPatchKindString, sourceCRs)
+
+	var err error
+	if sourceCRs != nil && *sourceCRs != "" {
+		err = fileutils.CopySourceCrs(*inputFile, *outputDir, preRenderSourceCRList)
+		if err != nil {
+			fmt.Printf("Could not copy source-crs files, err: %s", err)
+			os.Exit(1)
+		}
+	}
+
+	allFilesInInputPath, err := fileutils.GetAllYAMLFilesInPath(*inputFile)
+	if err != nil {
+		fmt.Printf("Could not get file list, err: %s", err)
+		os.Exit(1)
+	}
```
I think it would make sense to move this to the `processFlags` function.

edcdavid replied:
The processFlags function is only getting a list of kind and sourceCR path from the flags passed by the user. After this function all the info carried by the flags is usable. I added a comment in the code

edcdavid commented on ztp/gitops-subscriptions/argocd/example/acmpolicygenerator/README.md:
```diff
@@ -1,5 +1,78 @@
-The [ACM PolicyGenerator](https://github.com/stolostron/policy-generator-plugin) examples under [here](https://github.com/openshift-kni/cnf-features-deploy/tree/master/ztp/gitops-subscriptions/argocd/example/acmpolicygenerator) are defining the DU profile policies using [ACM PolicyGenerator reference API](https://github.com/stolostron/policy-generator-plugin/blob/main/docs/policygenerator-referen ...
+# ACM PolicyGenerator 
+The ACM Policy Generator Custom Resource (CRs) is the official template mechanism used to generate ACM policies. It is replacing Policy Generator Templates or PGT.
+ACM Policy Generator CRs are similar to PGT but there are few notable differences regarding its patching and placement strategies.
```
fixed

irinamihai commented:
Do we really need to copy files over here from open-cluster-management-io/policy-generator-plugin? Can't we include the policy-generator-plugin and directly use the things that we need?

irinamihai commented on ztp/tools/pgt2acm/test/schema.json:
```diff
@@ -0,0 +1,158 @@
+{
```
Maybe rename the file to something ~ `ptpconfig_schema.json`?

edcdavid replied:
Fixed. I deleted the schema.json.

irinamihai commented:
It looks like the PolicyGenerator and PolicyGenTemplate binaries are included in the commit.
Could we consider building the PolicyGenTemplate on the spot / copying it and bringing the PolicyGenerator from the ACM repo?

irinamihai commented:
I see that there are several source-crs used for testing purposes. Having 2 `source-crs` directories in this repo might be confusing.
I think the ideal approach would be to use the existing `source-crs` where possible and avoid duplicating them.

irinamihai commented on ztp/tools/pgt2acm/main.go:
```diff
@@ -0,0 +1,357 @@
+		}
+	}
+
+	if generateACMPolicies != nil && *generateACMPolicies {
```
For flags of `Bool` type I think it's enough to check `if *flag` since they all have defaults to `false`.

edcdavid replied:
Fixed

edcdavid commented:
/bugzilla refresh

edcdavid commented:
> It looks like the PolicyGenerator and PolicyGenTemplate binaries are included in the commit. Could we consider building the PolicyGenTemplate on the spot / copying it and bringing the PolicyGenerator from the ACM repo?

Fixed. I removed the executables, they were already generated on the spot, just also present in git.

edcdavid commented:
> I see that there are several source-crs used for testing purposes. Having 2 `source-crs` directories in this repo might be confusing. I think the ideal approach would be to use the existing `source-crs` where possible and avoid duplicating them.

Fixed. I removed the test source-crs. There is already a test that is using the existing source CR, so this one is redundant (it was useful in the standalone project)

sabbir-47 commented on ztp/tools/pgt2acm/main.go:
```diff
@@ -0,0 +1,359 @@
+		seenPoliciesSorted = append(seenPoliciesSorted, policyName)
+	}
+
+	sort.Strings(seenPoliciesSorted)
```
Why do we need to sort policy names?

edcdavid replied:
Yes, this is used later to consistently compare side by side rendered ACM policies from PGT and ACM PolicyGen

sabbir-47 commented on ztp/tools/pgt2acm/main.go:
```diff
@@ -0,0 +1,359 @@
+	}
+	relativeManifestPath, err := filepath.Rel(outputDir, renamedpathRelativeToOutputDir)
+	if err != nil {
+		return fmt.Errorf("cannot get the relative path from path: %s and directory: %s, err:%s", renamedpathRelativeToOutputDir, outputDir, err)
```
nit 
```suggestion
		return fmt.Errorf("cannot get the relative path from path: %s and directory: %s, err: %s", renamedpathRelativeToOutputDir, outputDir, err)
```

edcdavid replied:
Fixed

sabbir-47 commented on ztp/tools/pgt2acm/main.go:
```diff
@@ -0,0 +1,359 @@
+
+	renamedpathRelativeToOutputDir, err := fileutils.RenderMCPLines(pathRelativeToOutputDir, policyGenTemp.Spec.Mcp)
+	if err != nil {
+		return fmt.Errorf("cannot render MCP lines, err:%s", err)
```
nit
```suggestion
		return fmt.Errorf("cannot render MCP lines, err: %s", err)
```

edcdavid replied:
Fixed

sabbir-47 commented on ztp/tools/pgt2acm/main.go:
```diff
@@ -0,0 +1,359 @@
+				for manifestIndex := range acmGenTempConversion.Policies[policyIndex].Manifests {
+					err = RenderPatchesInManifestForSpecifiedKinds(&policyGenTemp, &acmGenTempConversion, policyIndex, manifestIndex, outputDir, schema, preRenderPatchKindList)
+					if err != nil {
+						return fmt.Errorf("could not render patches in manifest, err:%s", err)
```
nit
```suggestion
						return fmt.Errorf("could not render patches in manifest, err: %s", err)
```

edcdavid replied:
Fixed.

sabbir-47 commented on ztp/tools/pgt2acm/main.go:
```diff
@@ -0,0 +1,359 @@
+		fmt.Printf("Error writing to file, err: %s", err)
+		return err
+	}
+	fmt.Printf("Wrote converted ACM template: %s\n", outputFile)
```
Do we need this printf comment, can't we simply do?
`fmt.Println(outputFile)`

edcdavid replied:
Yes, we are not just printing the file name, there is a prefix string.

sabbir-47 commented on ztp/tools/pgt2acm/packages/patches/patches.go:
```diff
@@ -0,0 +1,292 @@
+	kustomizeDir        = "kustomize"
+)
+
+// Initializes the in-memory file system with base directory and open API schema
```
Please follow the convention of function name to start the description

edcdavid replied:
Fixed.

sabbir-47 replied:
I don't see it was fixed, what i meant was to start with the description with the function name.

sabbir-47 commented on ztp/tools/pgt2acm/packages/pgtformat/pgtformat.go:
```diff
@@ -0,0 +1,83 @@
+	type PolicyGenTemplateSpec PolicyGenTempSpec
+	var defaults = PolicyGenTemplateSpec{
+		WrapInPolicy:       true,     // Generate ACM wrapped policies by default
+		RemediationAction:  "inform", // Generate inform policies by default
```
define "inform" as constant

edcdavid replied:
Fixed

sabbir-47 commented on ztp/tools/pgt2acm/packages/patches/patches.go:
```diff
@@ -0,0 +1,292 @@
+	return nil
+}
+
+// Runs Kustomize to apply patches in the in-memory file system
```
follow the func name description convention

edcdavid replied:
Fixed

sabbir-47 replied:
I don't see it was fixed, what i meant was to start with the description with the function name

sabbir-47 commented on ztp/tools/pgt2acm/packages/labels/labels.go:
```diff
@@ -0,0 +1,67 @@
+	return selector
+}
+
+// outputs the Selector as a generic Yaml
```
follow the convention to describe func name. This is a comment in general. Please verify for all function name description.

edcdavid replied:
Fixed

sabbir-47 commented on ztp/tools/pgt2acm/packages/labels/labels.go:
```diff
@@ -0,0 +1,67 @@
+	return selector
+}
+
+// converts a labe list to a list of requirements in a selector
```
nit

edcdavid replied:
Fixed

sabbir-47 commented on ztp/tools/pgt2acm/packages/fileutils/fileutils.go:
```diff
@@ -0,0 +1,307 @@
+	if err != nil {
+		return fmt.Errorf("error writing to file: %s, err: %s", outputKustomization, err)
+	}
+	fmt.Printf("Wrote Updated Kustomization file: %s\n", outputKustomization)
```
this printf seems like commentary, please do a println

sabbir-47 commented on ztp/tools/pgt2acm/packages/fileutils/fileutils.go:
```diff
@@ -0,0 +1,307 @@
+	if err != nil {
+		return fmt.Errorf("error writing to file: %s, err: %s", fullNamespaceFilePath, err)
+	}
+	fmt.Printf("Added default placement binding to:%s\n", fullNamespaceFilePath)
```
nit
```suggestion
	fmt.Printf("Added default placement binding to: %s\n", fullNamespaceFilePath)
```

edcdavid replied:
Fixed

sabbir-47 commented on ztp/tools/pgt2acm/packages/fileutils/fileutils.go:
```diff
@@ -0,0 +1,307 @@
+	return annotations, nil
+}
+
+const defaultPlacementBindings = `
```
move this const top of the file where other constant are defined

edcdavid replied:
Fixed

sabbir-47 commented on ztp/tools/pgt2acm/packages/fileutils/fileutils.go:
```diff
@@ -0,0 +1,307 @@
+		if err != nil {
+			return err
+		}
+		//		if !info.IsDir() {
```
remove the comment

edcdavid replied:
Fixed

edcdavid commented:
/bugzilla refresh

edcdavid commented on ztp/tools/pgt2acm/main.go:
```diff
@@ -0,0 +1,359 @@
+
+	"flag"
+
+	"github.com/test-network-function/pgt2acm/packages/acmformat"
```
Fixed.

edcdavid commented on ztp/tools/pgt2acm/main.go:
```diff
@@ -0,0 +1,359 @@
+		// Convert Miscelanous fields
+		convertSimpleMiscellaneousFields(&policyGenTemp, &acmGenTempConversion, rootName)
+
+		if workaroundPlacement != nil && !*workaroundPlacement {
```
Fixed

edcdavid commented on ztp/tools/pgt2acm/packages/patches/patches.go:
```diff
@@ -0,0 +1,292 @@
+	return nil
+}
+
+// setPatchDefaults is a helper function for Validate that sets any missing values on the patches
```
Fixed.

edcdavid commented:
Some comments inline:

> * There are places where unit tests are needed to see the functions are returning value properly. I would recommend to add unit tests where it is required, especially at the package levels.
I can create another PR for this when this one merges

> * Please remove self commentary comments and only keep which are necessary
What do you me by self commentary ? paraphrasing the function name ?

rauhersu commented on ztp/resource-generator/Containerfile:
```diff
@@ -15,6 +16,11 @@ WORKDIR $PGT_ROOT
+# Build pgt2acm
+WORKDIR $PGT2ACM_ROOT
+RUN go mod vendor
+RUN make build
```
Does it make sense to encapsulate all the logic to build ('go mod vendor' in this case) inside the corresponding Makefile so 'make build' drives the build entirely? 

I have typically seen that components which claim to perform vendoring and are dockerized use a 'vendor' directory which is versioned and as part of the corresponding Dockerfile, a simple COPY of that directory is performed before building. Can we assess this approach ?

/cc @fontivan

edcdavid replied:
Fixed. Added a commit for it.

edcdavid commented:
/retest

edcdavid commented:
/retest

sabbir-47 commented on ztp/gitops-subscriptions/argocd/deployment/argocd-openshift-gitops-patch.json:
```diff
@@ -45,12 +45,12 @@
           "args": [
             "-c",
-            "mkdir -p /.config/kustomize/plugin/ && cp -r /etc/kustomize/plugin/policy.open-cluster-management.io /.config/kustomize/plugin/"
+            "mkdir -p /.config/kustomize/plugin/ && mkdir -p /.config/kustomize/plugin/policy.open-cluster-management.io/v1/policygenerator && cp /policy-generator/PolicyGenerator-not-fips-compliant /.config/kustomize/plugin/policy.open-cluster-management.io/v1/policygenerator/PolicyGenerator"
```
Why can't we omit the first mkdir?

irinamihai commented on ztp/gitops-subscriptions/argocd/example/acmpolicygenerator/README.md:
```diff
@@ -1,5 +1,77 @@
+   6. Commit the PolicyGenTemplate CRs, ns.yaml, and associated kustomization.yaml in git.
+4. Push your changes to the git repository, and the ArgoCD pipeline will detect the changes and begin the site deployment. The SiteConfig and PolicyGenTemplate CRs can be pushed simultaneously. Note: The policyGenTemplate CRs and associated ns.yaml, kustomization.yaml must be pushed to the git repository within the 20 mins after the SiteConfigs are pushed.
+
+## Adding Clustersetbinding 
```
ManagedClusterSetBinding

edcdavid replied:
fixed

irinamihai commented on ztp/tools/pgt2acm/README.md:
```diff
@@ -0,0 +1,761 @@
+
+This example assumes that the following is available:
+
+- a working gitopts repository. For instance, consider the following directory
```
gitops

edcdavid replied:
Fixed

irinamihai commented on ztp/tools/pgt2acm/README.md:
````diff
@@ -0,0 +1,761 @@
+KUSTOMIZE_PLUGIN_HOME=$(pwd)/kustomize 
+```
+
+so for instance, a example to generate policies for PGT and ACM Gen would be:
````
an example

irinamihai commented on ztp/tools/pgt2acm/README.md:
````diff
@@ -0,0 +1,761 @@
+The pgt2acm executable is also part of the official ZTP container. To run the pgt2acm tool run the following podmand command:
+
+```
+podman run --user root:root -e KUSTOMIZE_PLUGIN_HOME=/kustomize-pgt2acm -v $(pwd)/output:/home/ztp:Z -v $(pwd)/PolicyGenerator:/kustomize-pgt2acm/policy.open-cluster-management.io/PolicyGenerator:Z -v $(pwd)/policygentemplates:/policygentemplates:Z -v $(pwd)/acmgentemplates:/acmgentemplates:Z quay.io/openshift-kni/ztp-site-generator:latest  pgt2acm -i /policygentemplates -o /acmgentemplates  -c / ...
````
Maybe splitting such long commands would help with readability. 

```
podman run \
--user root:root \
-e KUSTOMIZE_PLUGIN_HOME=/kustomize-pgt2acm \
-v $(pwd)/output:/home/ztp:Z \
-v $(pwd)/PolicyGenerator:/kustomize-pgt2acm/policy.open-cluster-management.io/PolicyGenerator:Z \
-v $(pwd)/policygentemplates:/policygentemplates:Z \
-v $(pwd)/acmgentemplates:/acmgentemplates:Z \
quay.io/openshift-kni/ztp-site-generator:latest \
pgt2acm \
-i /policygentemplates \
-o /acmgentemplates \
-c /policygentemplates/source-crs -g
```

edcdavid replied:
fixed

irinamihai commented on ztp/tools/pgt2acm/main.go:
```diff
@@ -0,0 +1,421 @@
+	newPolicy.Name = rootName + "-" + policyName
+	newPolicy.PolicyAnnotations = make(map[string]string)
+	wave := ""
+	for srcFileIndex := range policyGenTemp.Spec.SourceFiles {
```
Can't this `for` be
```
for srcFileIndex, srcFile := range policyGenTemp.Spec.SourceFiles {
```
and then use `srcFile` throughout instead of always accessing it by using the index with `policyGenTemp.Spec.SourceFiles[srcFileIndex]`?

edcdavid replied:
This is a result from linting, the linter recommend using the index to avoid copying large  amount of data at each loop iteration (136 bytes)

edcdavid commented on ztp/gitops-subscriptions/argocd/deployment/argocd-openshift-gitops-patch.json:
```diff
@@ -45,12 +45,12 @@
             "/bin/bash"
           ],
-          "image": "registry.redhat.io/rhacm2/multicluster-operators-subscription-rhel8:v2.7",
+          "image": "registry.redhat.io/rhacm2/multicluster-operators-subscription-rhel9:v2.10",
```
I my understanding, the result of the discussion is that there a 2 policy-generator-plugin  executables in the RHEL9 multi cluster image, one supporting RHEL9 natively and another that is "universal", e.g. it supports RHEL 8 as well. It is ok to use the RHEL9 ACM image for now as long as we use the universal plugin executable for argoCD. When ArgoCD eventually moves to RHEL9 we will be able to use the RHEL9 image in the RHEL9 executable. ZTP will have to move to RHEL 9 at that time too. But for now we can stay on using RHEL8 for ZTP

sabbir-47 commented on ztp/tools/pgt2acm/main.go:
```diff
@@ -0,0 +1,421 @@
+			return err
+		}
+
+		// Convert Miscelanous fields
```
spelling error. Miscelanous -> miscellaneous

edcdavid replied:
fixed

irinamihai commented:
As a general comment, IMO, some of the names in the conversion tool are a bit inconsistent and would probably benefit from slight modifications. I suppose it could be done in a future PR. For example:
```
acmpolicyfrompgt -> acmpgfrompgt
acmgen-out.yaml > acmpg-out.yaml
ACM_POLICYGEN_FROM_PGT_DIR=./argocd/example/acmpolicyfrompgt -> ACMPG_FROM_PGT_DIR
test-acmgen-from-pgt -> test-acmpg-from-pgt
pgt2acm -> pgt2acmpg
```

irinamihai commented:
/lgtm

lack commented:
/override ci/prow/e2e-aws-ran-profile
/override ci/prow/e2e-aws-ci-tests

edcdavid commented:
/cc @schseba

lack commented:
/approve
